Empire/Empire.cpp

```diff
@@ -3386,6 +3388,20 @@ void Empire::UpdateOwnedObjectCounters() {
         if (!ship->SpeciesName().empty())
             m_species_ships_owned[ship->SpeciesName()]++;
         m_ship_designs_owned[ship->DesignID()]++;
+        // parts
+        const ShipDesign* design = ship->Design();
+        if (!design)
+            continue;
+        const std::vector<std::string>& parts = design->Parts();
```

Comment:
This might be better moved into ShipDesign. Could add a ShipDesign::PartCount function that returns, or takes in by reference and fills, maps from PartClass and from string to int? That info could also be cached in the ShipDesign in a mutable member after it is calculated...